### DISCUSSION TITLE
Open-RPC / JSON-RPC

### DISCUSSION THREAD

**Discussion by FraugDib** [Budibase/budibase#12779] [Ideas]
### The OpenRPC Specification defines a standard, programming language-agnostic interface description for [JSON-RPC 2.0](https://www.jsonrpc.org/specification) APIs. 

Also it's what's under the hood on just about every EVM (Ethereum Virtual Machine) cryptocurrency, yet somehow a reasonably well documented interface that doesn't require you to clone and build 1/2 dozen arcane github repos from 2019 is nowhere to be found.

**Comment by melohagan** [comment]
Hey @FraugDib What are you talking about?

**Reply by FraugDib** [reply]
This is a proposal to add [OpenRPC](https://github.com/open-rpc/spec) and [JSON-RPC] (https://www.jsonrpc.org/) data sources to Budibase so users can build tooling and  create powerful apps and workflows from these datasources and securely deploy professional-grade solutions across their teams. 

These protocols represent structural underpinnings of many (perhaps most) of the largest cryptocurrencies, and due to a severe lack of well documented tooling, along with the trustless nature of blockchain tech in general, the ability to create ones own tools with these data sources seems highly desirable. Budibase seems (IMO) like it would an ideal fit for this use case, were it to provide easy access to the JSON-RPC protocol (v2.0), and Open-RPC (a specification of JSON-RPC methods that an EVM-based blockchain client must implement).

Below are some informative links regarding OpenRPC and JSON-RPC

- The OpenRPC Document and generated Documentation that specifies all the methods an EVM-based blockchain should implement can be found [here](https://github.com/etclabscore/ethereum-json-rpc-specification).
- This was first proposed [here as an ECIP](https://github.com/etclabscore/ECIPs/blob/master/ECIPs/ECIP-1053.md), but the benefits of this kind of tooling is apparent across Bitcoin, Ethereum Classic, Ethereum and other JSON-RPC accessible blockchains.
- https://github.com/open-rpc/spec
- https://eips.ethereum.org/EIPS/eip-1901
- https://www.jsonrpc.org/
